src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,118 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Practitioner;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestStatus;
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestIntent;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private final ResourceFullUrlGenerator fullUrlGenerator;
+    private static final Map<ReportStatusCode, ProcedureRequestStatus> STATUS_CODE_MAPPING = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(
+        final Message message,
+        Patient patient,
+        Practitioner requestingPractitioner,
+        Organization requestingOrganization,
+        Practitioner performingPractitioner,
+        Organization performingOrganization
+    ) {
+        return message.getServiceReportDetails().getSubject().getClinicalInfo()
+            .map(clinicalInfo -> mapPatientClinicalInfo(clinicalInfo, patient, requestingPractitioner,
+                requestingOrganization, performingPractitioner, performingOrganization));
+    }
```

Comment:
A lot of the complication of this class is from having to pass around these variables all the time, which you have to do because this is a singleton `@Component` and it can't hold state. There are some alternatives we could try though:

1. Where this class is autowired, replace it with a `ProcedureRequestMapperFactory` that accepts all of the other mapped vars and returns a one-use/one-thread `ProcedureRequestMapper` that can use those vars as its state and won't need to keep passing them around.

2. Keep the current autowiring and entry method so they all keep the same "look", but this class itself will just spin up another object that will actually do the work; statefully, as in option 1.
2a. Use an inner class
2b. Use a sibling class

3. Final option from me is to explore the use of `ThreadLocal` so maybe you don't need another class, but can still let this one hold state, per thread. I was going to look into it for the `ObservationMapper` but it ended up being unnecessary.

(And 4: if you don't see any problem, keep it as it is!)

Reply:
Will it better in this way? ca3b95096a23c49297a51a1c21e36d4b6b74e421
Thanks!